src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -26,28 +280,155 @@ static void subCacheItemUpdate(const char* tenant, BSONObj* subP)
   }
 
 
-  std::string               subId             = idField.OID().toString();
-  std::string               servicePath       = subP->hasField(CSUB_SERVICE_PATH)? subP->getField(CSUB_SERVICE_PATH).String() : "/";
-  std::string               reference         = subP->getField(CSUB_REFERENCE).String();
-  int64_t                   expiration        = subP->hasField(CSUB_EXPIRATION)? subP->getField(CSUB_EXPIRATION).Long() : 0;
-  int64_t                   throttling        = subP->hasField(CSUB_THROTTLING)? subP->getField(CSUB_THROTTLING).Long() : -1;
+  //
+  // 03. Create CachedSubscription
+  //
+  CachedSubscription* cSubP = (CachedSubscription*) calloc(1, sizeof(CachedSubscription));
+
+  if (cSubP == NULL)
+  {
+    LM_E(("Runtime Error (cannot allocate memory for a cached subscription: %s)", strerror(errno)));
```

Comment:
Probably this should be Fatal Error and invoke a orionExit(). The broker shouldn't continue running if memory has been exahusted.

However, don't need to fix this right now. Just use a "FIXME" above this line to indicate that behaviour for a next round.